### PR TITLE
Stdlib consistency

### PR DESCRIPTION
**Pull request by phated**
These are some breaking changes to the stdlib that:
* Renames some methods to make things more clear (`foldLeft` -> `reduce`, `hd` -> `head`, etc)
* Updates some methods that didn't adhere to the collection-last style
* Prefers longer variable names to make the stdlib easier for non-FP people to contribute

After this is approved, I will get all the docs changed on the website before merging.

### CODE REVIEW

File: compiler/test/stdlib/lists.test.gr [hunk 1]
```diff
@@ -124,6 +124,6 @@ assert rotate(-2, list) == [2, 3, 1]
 
 # List.uniq
```
Comment (ospencer) [inline compiler/test/stdlib/lists.test.gr:125]:
```suggestion
# List.unique
```

File: stdlib/lists.gr [hunk 2]
```diff
@@ -2,185 +2,196 @@
...
 
-let part = (n, l) => {
-  if (n < 0) fail 'part'
-  let rec help = (l1, l2, n) => {
-    match (l2) {
-      | [] => if (n > 0) fail 'part' else (l1, l2)
-      | [hd, ...tl] => if (n > 0) help([hd, ...l1], tl, n - 1) else (l1, l2)
+let part = (count, list) => {
+  if (count < 0) {
+    fail 'Cannot be a negative number'
```
Comment (ospencer) [inline stdlib/lists.gr:171]:
```suggestion
    fail 'part count cannot be a negative number'
```

File: stdlib/lists.gr [hunk 3]
```diff
@@ -2,185 +2,196 @@
...
   }
 }
 
-let rec insert = (x, i, l) => {
-  match(l) {
-    | [] => if (i == 0) [x] else fail 'insert'
-    | [hd, ...tl] => if (i < 0) fail 'insert' else if (i == 0) [x, ...l] else [hd, ...insert(x, i - 1, tl)]
+let rec insert = (value, index, list) => {
+  if (index < 0) {
+    fail 'Cannot be a negative number'
```
Comment (ospencer) [inline stdlib/lists.gr:153]:
```suggestion
    fail 'insert index cannot be a negative number'
```

File: stdlib/lists.gr [hunk 4]
```diff
@@ -2,185 +2,196 @@
...
-    | [] => fail 'nth'
-    | [hd, ...tl] => if (n < 0) fail 'nth' else if (n == 0) hd else nth(n - 1, tl)
+let rec nth = (count, list) => {
+  if (count < 0) {
+    fail 'Cannot be a negative number'
+  } else {
+    match (list) {
+      | [] => fail 'Cannot get any elements from an empty List'
+      | [first, ...rest] => if (count == 0) first else nth(count - 1, rest)
+    }
```
Comment (ospencer) [inline stdlib/lists.gr:140]:
```suggestion
let rec nth = (index, list) => {
  if (index < 0) {
    fail 'nth index cannot be a negative number'
  } else {
    match (list) {
      | [] => fail 'Cannot get any elements from an empty List'
      | [first, ...rest] => if (index == 0) first else nth(count - 1, rest)
    }
```

**Comment by phated** [comment]
@ospencer can you give that updates commit another check? I updated some other error messages to mention out-of-bounds after thinking more about them while writing docs.

File: stdlib/option.gr [hunk 5]
```diff
@@ -1,69 +1,69 @@
-# Standard library for Option functionality
+# Standard library for optionion functionality
```
Comment (ospencer) [inline stdlib/option.gr:1]:
```suggestion
# Standard library for option functionality
```

**Pull request merged**